### ISSUE TITLE
improve AENS auction system

### ISSUE THREAD

**Issue by marc0olo** [aeternity/aeternity#3853]
I had lately some discussions again about the way the auctions work and I searched for old forum entries about that, see:
- https://forum.aeternity.com/t/aeternity-nft-token-standard/9781/7
- https://forum.aeternity.com/t/aens-name-expiration-time-time-to-make-a-vote/8063/20

inviting @arjanvaneersel to participate in the discussion here.

I am citing his thoughts here:
- "The duration of the auctions is way too long, you can’t seriously expect parties to wait for and x amount of days or weeks to get their name registered"
- "The whole idea of auction seems to have been introduced against name squatting, but in reality it’s not much of a protection against this, it only drives the price up"
- "Even if it would be effective against name squatting it opens up a potential other form of abuse, which is IMHO even worse than squatting: The possibility to drive up the price through bidding wars."

my personal biggest concern is how the auction system works **AFTER** placing the first bid. from my observations it it like this:
- I claim `abcdefghijk.chain` (auction starts for 24 hours)
- 23 hours pass
- somebody bids for `abcdefghijk.chain` (auction duration is bumped to 24 hours again)
- 23 hours pass
- somebody bids for `abcdefghijk.chain` (auction duration is bumped to 24 hours again)
- ...

I heavily dislike this behavior. IMO we should **at least** change this behavior for every bid that comes after the first bid. we could bump the duration e.g. only to 2-5 keyblocks if it's within the 5 keyblocks until the auction ends.

maybe we find other ways to improve, too. but currently auctions can basically run forever and one needs to wait ages to claim their desired name

**Comment by keypair** [comment]
For me 24h extension seems fine if a higher bid arrives so that people don't bid last minute and take away names from others. But extending the auction for a year isn't acceptable IMO.

**Comment by sunbx** [comment]
I personally feel that the current bidding logic is a bit painful and if someone raises the price it will start from scratch.
I wish it would be more reasonable, I claim A.Chain There is a three-month public notice period after registration
Anyone can bid up during this period
If the bidding period is less than 24 hours, the bidding period is automatically recalculated from 24 hours

**Comment by marc0olo** [comment]
to me even 24 hours seem too much. but every change in that aspect would be a massive improvement.

I think 5 keyblocks (~ 15mins) is fine. even 1, 2 or 3 keyblocks should be sufficient if there is demand to get the name.

**Comment by sunbx** [comment]
> to me even 24 hours seem too much. but every change in that aspect would be a massive improvement.
> 
> I think 5 keyblocks (~15 mins) is fine. even 1, 2 or 3 keyblocks should be sufficient if there is demand to get the name.

If the time is too short, there may be time difference or other factors of not being online, which may cause the domain name to be aucted by others last and lose the domain name you like

**Comment by marc0olo** [comment]
> If the time is too short, there may be time difference or other factors of not being online, which may cause the domain name to be aucted by others last and lose the domain name you like

true, but the time is then known to everybody interested in it. if I wanna watch the SuperBowl I am also having a hard night 😅

**Comment by jdgcs** [comment]
Thanks @[sunbx](javascript:;) that he mentioned this discuss in the Wechat group. We discussed a lot from the business model of the AENS auction. In brief:

**1. Return a rate of the added price to the last bidder**. eg, A started a new auction: 352 AE for name 1.chain, B added the price to 452AE, the system might return 80% of the added price((452-353)*0.8=80 AE) to A, and 20%((452-353)*0.2=20 AE) to BRI. This mechanism might raise the enthusiasm for the bidders, and more income for BRI. For the bidder who want the name, he lost nothing comparing the current mechanism.(From **Maling** and me).

**2. Protect the first bidder's privilege.** With a more aggressive model, the bidding mechanism might return a rate(5%,maybe) of the added price to the first bidder. So the bidders might have a very high motivation to spread AENS and his auctions,(From **XBanxian**).

As mentioned before, aeknow.chain(https://github.com/aeknow/aeks) with AENS website and Chaet is in polishing, the beta version will be released in recent. It could be a good application and story for the AENS.

If possible, the accurate return rate parameter could be discussed deeper.

**Comment by hanssv** [comment]
This is the category of changes that are more or less trivial to do implementation-wise, but where reaching a consensus is the hard work. Sounds like something that should be voted upon perhaps?!

**Comment by marc0olo** [comment]
> Sounds like something that should be voted upon perhaps?!

what options do we have implementation-wise? what can you propose to solve this issue short term? is it about the conditions of "bumping" the remaining time in case of last minute bids?

I also heard the opinion of @arjanvaneersel in some place to get completely rid of the auction system. from a company's perspective it might not be convenient to wait for one or multiple months just to get a name.

I have no strong opinions about this. personally I really like the auction system in general.

**Comment by hanssv** [comment]
@marc0olo Nothing can be done short term since the auction is part of the consensus protocol. So ~~hard-fork~~ protocol upgrade needed.

Different schemes with regards to auction extension and payback distribution is relatively straightforward to implement.

**Comment by arjanvaneersel** [comment]
@marc0olo, @hanssv: I have thought a bit more about this and I stick to my objections to the auction system from a business perspective, which are in short:

- The auction could theoretically prevent squatting of names, but it also opens the door to bidding wars to drive the prices up. This is especially nasty if someone would try to register a company name or trademark;
- In the case of registering things that are typical for a particular business entity, like a registration number, company name or trademark it makes absolutely no sense that registration takes so long. Event though this might be easy to fix I still see no benefit of having an auction for names based on unique properties of an entity.

Perhaps the auction system makes sense with private individuals even though I also see no benefits there, but if the majority believes this is the right way then I will accept that of course.

So perhaps it would be an idea to implement different tlds with different policies for these different use cases.

**Comment by marc0olo** [comment]
different tlds in general is definitely interesting. but there also the question remains => how should they be claimable / distributed? how could we enforce such policies (which need to be defined) on protocol level? maybe we could have some sort of DAO that can make decisions if a domain / tld should be handed out or not

how would you prevent squatting without any protection mechanism @arjanvaneersel? or don't you think there should be some way to prevent it?

**Comment by arjanvaneersel** [comment]
@marc0olo Neither squatting nor bidding wars are ideal. But I do consider squatting the lesser of the two evils. A come first, be served first principle is not bad, especially when a name also expires. I would have less of a problem with auctions if the name wouldn't expire.

I didn't go into the governance of tlds on purpose, because there are several options. I also thought about a dedicated DAO. The governance itself would depend on the purpose of the tld. In my particular purpose it is business oriented, so the procedure could be that a business or trademark is verified for registration. However, that would require manual interaction from DAO members, as not every trade register can be accessed online. 

DAO members who'll do the unavoidable manual labor could get rewarded for their work by getting a share of the registration fee. It's also not unfair is that fee is higher than the fee for an individual's .chain registration. Afterall, we are offering a solid form of business verification here. So when parties working with an AENS name from the business TLD they get certainty that this entity is duly verified. I believe that will increase confidence for doing business on-chain.

The other option is to have some trust in people who register a name and that verification would take place only if someone objects. I think that the risk of losing the name might be enough of motivation for people to play honest, because in this use case it's not hard to detect abuse.

For other tlds this governance model might not work, so -as I said- it all depends on the exact TLD.

**Comment by marc0olo** [comment]
we started a discussion in the forum:
- https://forum.aeternity.com/t/discussion-improvement-of-the-auction-system-for-ceres-hardfork/10907
